src/include/sof/audio/volume.h

```diff
@@ -162,7 +187,29 @@ static inline vol_scale_func vol_get_processing_function(struct comp_dev *dev)
 
 	return NULL;
 }
+#else
+/**
+ * \brief Retrievies volume processing function.
+ * \param[in,out] dev Volume base component device.
+ */
+static inline vol_scale_func vol_get_processing_function(struct comp_dev *dev)
+{
+	struct vol_data *cd = comp_get_drvdata(dev);
+
+	switch (cd->base.audio_fmt.depth) {
+	case IPC4_DEPTH_16BIT:
+		return func_map[0].func;
+	case IPC4_DEPTH_32BIT:
+		return func_map[2].func;
+	default:
+		comp_err(dev, "vol_get_processing_function(): unsupported depth %d",
+			 cd->base.audio_fmt.depth);
+		return NULL;
+	}
 
+	return NULL;
```

Comment:
one of the two `return NULL;` statements is really superfluous, either of them alone would suffice and even the compiler wouldn't mind. Pick your favourite.